### PR TITLE
Fixes a good few improper overlay icon_states, fuck gun code edition

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Fixes a lot of improper overlay creation, sussed out by #59553 and my hell project
## Why It's Good For The Game
Less runtimes in future, I'll have an easier time getting a thing done

### CODE REVIEW

File: code/modules/projectiles/guns/ballistic/rifle.dm [hunk 1]
```diff
@@ -193,6 +189,7 @@
 	item_flags = NEEDS_PERMIT | DROPDEL | ABSTRACT | NOBLUDGEON
 	flags_1 = NONE
 	trigger_guard = TRIGGER_GUARD_ALLOW_ALL
+	show_bolt_icon = FALSE //It's a fucking why is this a gunnnnnnnnnnnn
```
Comment (ShizCalev) [inline code/modules/projectiles/guns/ballistic/rifle.dm:192]: english?

File: code/modules/projectiles/guns/ballistic.dm [hunk 2]
```diff
@@ -72,6 +72,8 @@
 	var/casing_ejector = TRUE
 	///Whether the gun has an internal magazine or a detatchable one. Overridden by BOLT_TYPE_NO_BOLT.
 	var/internal_magazine = FALSE
+	///Should we attempt to display the gun's magazine?
+	var/display_mag = TRUE
```
Comment (ShizCalev) [inline code/modules/projectiles/guns/ballistic.dm:76]: is it really necessary add an addition check for **_every single time a gun is fired_** when only a single meme weapon utilizes it? why not just add an invisible icon for the stickman gun instead?

Comment (LemonInTheDark) [reply code/modules/projectiles/guns/ballistic.dm:76]: if I'm honest I thought I'd find more like this, I'll pull the var and add a invisible icon

Comment (LemonInTheDark) [reply code/modules/projectiles/guns/ballistic.dm:76]: wait a sec this  var already exists, it's just totally unused. wtf

File: code/modules/projectiles/guns/ballistic.dm [hunk 3]
```diff
@@ -181,7 +181,7 @@
 		. += "[icon_state]_toy"
 
 
-	if(!magazine || internal_magazine)
+	if(!magazine || internal_magazine || !mag_display)
```
Comment (ShizCalev) [inline code/modules/projectiles/guns/ballistic.dm:184]: honestly i still think removing this var and adding an invisible `flatgun_mag` icon is a far better option over adding a snowflake check that's run every single time a ballistic gun fires just for a single meme spaceruin weapon.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/ballistic.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/modules/projectiles/guns/ballistic.dm:184]: my concern is more stuff that doesn't want to display mags popping up, and then people duplicate the solution, or worse, add the var themselves and don't know to update the flatgun

Comment (LemonInTheDark) [reply code/modules/projectiles/guns/ballistic.dm:184]: If you just want it gone I can do that

**Pull request merged**